typing/ctype.ml

```diff
@@ -3366,8 +3375,13 @@ let add_method env label priv virt ty sign =
     | (priv', virt', ty') -> begin
         let priv =
           match priv' with
-          | Public -> Public
-          | Private -> priv
+          | Mpublic -> Mpublic
+          | Mprivate k ->
+            match priv with
+            | Public ->
+                link_kind ~inside:k field_public;
```

Comment:
This can raise an `Invalid_argument` exception inside `link_kind` if the same method is added multiple times. For example, this crashes the toplevel
```ocaml
class foo =
  object
    method private f (b : bool) = b
    inherit object
      method f (b : bool) = b
    end
  end;;
```